iOS/Sources/Beagle/Sources/Navigation/BeagleNavigator.swift

```diff
@@ -66,12 +66,17 @@ class BeagleNavigator: BeagleNavigation {
     // MARK: - Navigate Handle
     
     private func navigate(route: Route, origin: BeagleController, animated: Bool, transition: @escaping Transition) {
-        viewController(route: route, origin: origin, retry: {
-            [weak origin] in guard let origin = origin else { return }
-            self.navigate(route: route, origin: origin, animated: animated, transition: transition)
-        }) {
-            transition(origin, $0, animated)
-        }
+        viewController(
+            route: route,
+            origin: origin,
+            retry: {
+                [weak origin] in guard let origin = origin else { return }
```

Comment:
Just a suggestion, for me is more clear.

```suggestion
            retry: { [weak origin] in
                guard let origin = origin else { return }
```

Reply:
done 😁